azkaban-common/src/main/java/azkaban/utils/SystemMemoryInfo.java

```diff
@@ -0,0 +1,165 @@
+package azkaban.utils;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.log4j.Logger;
+
+/**
+ * @author wkang
+ * This class is used to maintain system memory information. Processes utilizing
+ * large amount of memory should consult this class to see if the system has enough
+ * memory to proceed the operation.
+ * 
+ * Memory information is obtained from /proc/meminfo, so only Unix/Linux like system
+ * will support this class.
+ * 
+ * All the memory size used in this function is in KB
+ */
+public class SystemMemoryInfo {
```

Comment:
It will be a good idea to abstract SystemMemoryInfo implementation from the caller. Just wondering.. should we make it an interface SystemMemoryInterface and have SystemMemoryInfo implement it so later on we can change implementation without affecting client code or even use reflection to instantiate class reading conf ?